contexts/compound-and-earn-context.js

```diff
@@ -82,8 +72,34 @@ export function CompoundAndEarnProvider({ children }) {
     }
   }
 
+  const getBalanceInfosByPool = async () => {
+    if (!account) {
+      setPopUp({
+        title: 'Network Error',
+        text: `Please Switch to Avalanche Chain and connect metamask`,
+      });
+      return false;
+    } else {
+      const dataWithPollBalance = await Promise.all(
+        data?.LastSnowballInfo?.poolsInfo.map(async (item) => {
+          const gaugeContractForPNG = new ethers.Contract(item.address, GAUGE_ABI, library.getSigner());
```

Comment:
you can use the same code for png and joe pools
const gaugeContract = new ethers.Contract(item.gaugeInfo.address, GAUGE_ABI, library.getSigner());